src/gpt.py

```diff
@@ -113,10 +114,19 @@ def parse_llmresult(self, llmresult: AIMessage) -> Dict[str, Dict]:
 
 
 class GPTAnswerer:
-    def __init__(self, openai_api_key):
-        self.llm_cheap = LoggerChatModel(
-            ChatOpenAI(model_name="gpt-4o-mini", openai_api_key=openai_api_key, temperature=0.4)
-        )
+    def __init__(self, config, llm_api_key):
+        llm_model_type = config['llm_model_type']
+        llm_model = config['llm_model']
+        llm_api_url = config['llm_api_url']
+
+        print('Using {0} with {1} from {2}'.format(llm_model_type, llm_model, llm_api_url))
+
+        if llm_model_type == "ollama":
```

Comment:
I think you can remove this if by recommending that people use the ollama openai API compatible endpoint https://github.com/ollama/ollama/blob/main/docs/openai.md

also allows any other openai compatible provider to be used (there are many), also via something like https://docs.litellm.ai/docs/proxy/quick_start

Reply:
This feature is experimental as mentioned on the site you provided:
`Ollama provides experimental compatibility with parts of the [OpenAI API](https://platform.openai.com/docs/api-reference) to help connect existing applications to Ollama`